File: develop-api/src/main/java/com/salesianos/triana/backend/Animangav4/dtos/EditUserDto.java
package com.salesianos.triana.backend.Animangav4.dtos;

import com.salesianos.triana.backend.Animangav4.validation.annotation.EditEmail;
import lombok.*;

import javax.validation.constraints.Email;
import javax.validation.constraints.NotBlank;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Builder
@EditEmail(
        usernameField = "username",
        emailField = "email",
        message = "{user.unique.email}")
public class EditUserDto {

    private String username;
    private String fullName;
    @NotBlank(message = "{user.email.blank}")
    @Email(message = "{user.email.email}")
    private String email;
}

File: develop-mobile/animangav5frontend/lib/models/edit_user_dto.dart
class EditUserDto {
  EditUserDto({
    required this.fullName,
    required this.email,
    required this.username,
  });
  late final String fullName;
  late final String email;
  late final String username;

  Map<String, dynamic> toJson() {
    final _data = <String, dynamic>{};
    _data['fullName'] = fullName;
    _data['email'] = email;
    _data['username'] = username;
    return _data;
  }
}
